sycl/source/detail/sycl_mem_obj_t.cpp

```diff
@@ -223,6 +223,19 @@ void SYCLMemObjT::detachMemoryObject(
     Scheduler::getInstance().deferMemObjRelease(Self);
 }
 
+void SYCLMemObjT::handleWriteAccessorCreation() {
+  const auto InitialUserPtr = MUserPtr;
+  MCreateShadowCopy();
+  MCreateShadowCopy = []() -> void {};
+  if (MRecord != nullptr && MUserPtr != InitialUserPtr) {
+    for (auto &it : MRecord->MAllocaCommands) {
+      if (it->MMemAllocation == InitialUserPtr) {
+        it->MMemAllocation = MUserPtr;
+      }
+    }
+  }
```

Comment:
Tiny nit to avoid a bit of indentation. If you prefer the current version I am also okay with sticking with it.
```suggestion
  if (!MRecord || MUserPtr == InitialUserPtr)
    return;
  for (auto &it : MRecord->MAllocaCommands)
    if (it->MMemAllocation == InitialUserPtr)
      it->MMemAllocation = MUserPtr;
```